src/main/java/org/folio/entlinks/service/authority/AuthoritySourceFileService.java

```diff
@@ -87,9 +98,14 @@ public AuthoritySourceFile update(UUID id, AuthoritySourceFile modified) {
     }
 
     var existingEntity = repository.findById(id).orElseThrow(() -> new AuthoritySourceFileNotFoundException(id));
+    if (modified.getVersion() < existingEntity.getVersion()) {
+      throw OptimisticLockingException.optimisticLockingOnUpdate(
+          id, existingEntity.getVersion(), modified.getVersion());
+    }
 
-    copyModifiableFields(existingEntity, modified);
+    updateSequenceStartNumber(existingEntity, modified);
```

Comment:
I thought that it would be better to update the sequence after the source file record is updated as these are two separate transactions and the success of one (sequence) depends on the success of another (source file), so that not to have the situation where update finished with failure (for instance optimistic lock) but we have updated sequence. 
For this reason I was also separating the creation of source file and sequence into two steps.

Reply:
What do you mean separate transactions? All three operations "repository.findById", "updateSequenceStartNumber", "repository.save" are executed inside the same method "update" which is annotated with "@Transactional" in order to have all three operations inside one transaction

Reply:
Tested locally with docker-compose:
- on Optimistic locking error nothing is updated because it's thrown before updating anything
- if test error throw added after repository.save -> both operations got rollbacked - updating sequence and entity save so nothing is updated